hasher-matcher-actioner/tests/scripts/check_deployed_instance.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Utility script to confirm the basic functionally of a deployed hma instance
+"""
+
+import typing as t
+import logging
+import os
+import boto3
+from botocore.exceptions import ClientError
+import time
+import sys
+
+s3_client = boto3.client("s3")
+dynamodb = boto3.resource("dynamodb")
+
+# Copy of what is configured in tf files (not worth building a parser for them for single point of mataince imo)
+THREAT_EXCHANGE_PDQ_DATA_KEY = "threat_exchange_data/pdq.te"
+PDQ_INDEX_KEY = "index/pdq_hashes.index"
+
+# Test uses hasher-matcher-actioner/tests/data/b.jpg
+TEST_PHOTO_KEY = "images/test_photo.jpg"
+TEST_PHOTO_EXPECTED_HASH = (
+    "f8f8f0cee0f4a84f06370a22038f63f0b36e2ed596621e1d33e6b39c4e9c9b22"
+)
+TEST_PHOTO_EXPECTED_ID = "5555555555555555"
+
+TEST_TE_DATA_PDQ = f"""0000000000000000000000000000000000000000000000000000000000000000,0000000000000001,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+000000000000000000000000000000000000000000000000000000000000ffff,0000000000000001,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f,0000000000000002,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0,0000000000000003,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff,0000000000000004,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+1111111111111111111111111111111111111111111111111111111111111111,0000000000000005,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+{TEST_PHOTO_EXPECTED_HASH},{TEST_PHOTO_EXPECTED_ID},2020-07-31T18:47:52+0000,tag1 tag2 tag3
+"""
+
+
+class TestError(Exception):
+    """
+    Wrapper for exceptions which cause return codes
+    """
+    def __init__(self, message: str, returncode: int = 1) -> None:
+        super().__init__(message)
+        self.returncode = returncode
+
+
+def upload_s3_object(bucket: str, key: str, body: str) -> None:
+    """
+    Uploads an object,
+    throws if upload fails (put_object only returns if succesful).
+    """
+    try:
+        print(f"Attempting to upload {key}...")
+        s3_client.put_object(Bucket=bucket, Key=key, Body=body)
+    except ClientError as e:
+        raise TestError(e.msg)
+
+
+def wait_for_s3_object(bucket: str, key: str, wait_time=5, retries=20) -> None:
+    """
+    Waits for specfic object in given s3 bucket.
+    """
+    try:
+        print(f"Looking for {key} in {bucket} (check every {wait_time}s x{retries})...")
+        waiter = s3_client.get_waiter("object_exists")
+        waiter.wait(
+            Bucket=bucket,
+            Key=key,
+            WaiterConfig={"Delay": wait_time, "MaxAttempts": retries},
+        )
+    except ClientError as e:
+        raise TestError(e.msg)
+
+
+def wait_for_db_item(
+    table, key: t.Dict, attributes: t.List, wait_time=5, retries=20
+) -> t.Dict:
+    """
```

Comment:
Why do we do this in a loop. Why might it not be there first time?

Reply:
The lambda needs time to do its job and for things to propagate as both uploads trigger an event the call the lambdas